include/SFML/Network/Http.hpp

```diff
@@ -173,7 +173,7 @@ class SFML_NETWORK_API Http : NonCopyable
         ////////////////////////////////////////////////////////////
         // Types
         ////////////////////////////////////////////////////////////
-        typedef std::map<std::string, std::string> FieldTable;
+        typedef std::unordered_map<std::string, std::string> FieldTable;
```

Comment:
This also changes semantics.
https://github.com/SFML/SFML/blob/eef309e870f523363339e02b95d4f6eb2a18b429/src/SFML/Network/Http.cpp#L121-L125

The HTTP request now has a different payload.

What is even worse however is that `unordered_map` has no ordering guarantees whatsoever, so the same SFML code run on different stdlib implementations will produce different HTTP payloads. This might even happen on the same machine in different run-throughs, not sure if `unordered_map` has deterministic ordering or hash DoS protection.

I would revert this change and write a comment why we need `std::map`'s ordering, to make sure it's not refactored in the future.

Reply:
I realized that this would lead to different ordering, but is that actually a problem? The HTTP payload seems to rely on a key-value based approach, why does the ordering matter? 

I assumed that the associative nature of the payload would mean that it would still work -- am I wrong?

Reply:
> I realized that this would lead to different ordering, but is that actually a problem? The HTTP payload seems to rely on a key-value based approach, why does the ordering matter?

The main problem is that it's not deterministic. The same SFML code can lead to different HTTP payloads.

While I agree that _in theory_ neither HTTP nor the server code _should_ care about the order, in practice we don't know what exactly the other end does. Even something simple as logging HTTP requests and finding diffs between them for debugging purposes becomes an issue.

Since there's no advantage in using `unordered_map` here, I'd go with a more user-friendly behavior and potentially save some unnecessary debugging for poor developers 🙂

Reply:
I am somewhat convinced. I will revert this particular change and add a comment.